jwtmiddleware.go

```diff
@@ -160,58 +220,35 @@ func FromFirst(extractors ...TokenExtractor) TokenExtractor {
 	}
 }
 
-func (m *JWTMiddleware) CheckJWT(w http.ResponseWriter, r *http.Request) error {
-	if !m.Options.EnableAuthOnOptions {
-		if r.Method == "OPTIONS" {
+func (m *JWTMiddleware) CheckJWT(r *http.Request) error {
+	if !m.validateOnOptions {
+		if r.Method == http.MethodOptions {
 			return nil
 		}
 	}
 
-	// Use the specified token extractor to extract a token from the request
-	token, err := m.Options.Extractor(r)
-
-	// If debugging is turned on, log the outcome
+	token, err := m.tokenExtractor(r)
 	if err != nil {
-		m.logf("Error extracting JWT: %v", err)
```

Comment:
note that we have completely dropped debug functionality for several reasons:
- in some places we were logging the full string of the JWT 😱 and that is unacceptable. 
- as a library it's more idiomatic to pass errors up and let those be logged by the package consumer instead of logging them in the package